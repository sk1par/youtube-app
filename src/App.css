*, body, .ui.segment {
background: #f2709c;  /* fallback for old browsers */
}

.video-item {
  display: flex !important;
  align-items: center !important;
  cursor: pointer;
}

.video-item.item img {
  max-width: 200px;
  max-height: 250px ;
}

.button {
  margin-top: 5px !important;
  display: flex;
  justify-content: center;
}
