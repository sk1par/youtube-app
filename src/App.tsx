import React, { useState } from 'react';
import './App.css';
import youTubeApi from './api/youtube';
import SearchBar from './components/SearchBar';
import VideoDetail from './components/VideoDetail';
import VideoList from './components/VideoList';
import IVideos from './models/videos.interface';
import { AxiosResponse } from 'axios';

function App() {
  const [videos, setVideos] = useState<IVideos[]>([]);
  const [selectedVideo, setSelectedVideo] = useState<IVideos>();

  const handleSubmit = async (termFromSearchBar: string) => {
    const response= await youTubeApi.get('/search', {
        params: {
            q: termFromSearchBar
        }
    })

    setVideos(response.data.items);
};

  const handleVideoSelect = (video: IVideos) => {
    console.log(video);
    setSelectedVideo(video);
}

  return (
    <div className='ui container' style={{ marginTop: '1em' }}>
      <SearchBar handleFormSubmit={handleSubmit} />
      <div className='ui grid'>
        <div className="ui row">
          <div className="eleven wide column">
            <VideoDetail video={selectedVideo} />
          </div>
          <div className="five wide column">
            <VideoList handleVideoSelect={handleVideoSelect} videos={videos} />
          </div>
        </div>
      </div>
    </div>
  );
}

export default App;
