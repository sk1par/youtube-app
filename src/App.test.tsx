import App from './App';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

let wrapper: any;

beforeEach(() => {
    wrapper = shallow(<App />);
});

it('renders correctly', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
