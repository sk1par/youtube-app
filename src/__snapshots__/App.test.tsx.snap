// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1em",
    }
  }
>
  <SearchBar />
  <div
    className="ui grid"
  >
    <div
      className="ui row"
    >
      <div
        className="seven wide column"
      >
        <VideoDetail />
      </div>
      <div
        className="four wide column"
      >
        <History />
      </div>
      <div
        className="five wide column"
      >
        <VideoList />
      </div>
    </div>
  </div>
</div>
`;
